frontend/components/EnrollSecretTable/EnrollSecretRow/EnrollSecretRow.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { useState } from "react";
+
+import Button from "components/buttons/Button";
+// @ts-ignore
+import InputField from "components/forms/fields/InputField";
+// @ts-ignore
+import FleetIcon from "components/icons/FleetIcon";
+// @ts-ignore
+import { stringToClipboard } from "utilities/copy_text";
+import { IEnrollSecret } from "interfaces/enroll_secret";
+import EyeIcon from "../../../../assets/images/icon-eye-16x16@2x.png";
+import EditIcon from "../../../../assets/images/icon-pencil-14x14@2x.png";
+import DeleteIcon from "../../../../assets/images/icon-trash-14x14@2x.png";
+
+const baseClass = "enroll-secrets";
+
+interface IEnrollSecretRowProps {
+  secret: IEnrollSecret;
+  toggleSecretEditorModal?: () => void;
+  toggleDeleteSecretModal?: () => void;
+  setSelectedSecret?: React.Dispatch<
+    React.SetStateAction<IEnrollSecret | undefined>
+  >;
+}
+const EnrollSecretRow = ({
+  secret,
+  toggleSecretEditorModal,
+  toggleDeleteSecretModal,
+  setSelectedSecret,
+}: IEnrollSecretRowProps): JSX.Element | null => {
+  const [showSecret, setShowSecret] = useState<boolean>(false);
+  const [copyMessage, setCopyMessage] = useState<string>("");
+
+  const onCopySecret = (evt: React.MouseEvent) => {
+    evt.preventDefault();
+
+    stringToClipboard(secret.secret)
+      .then(() => setCopyMessage("Copied!"))
+      .catch(() => setCopyMessage("Copy failed"));
+
+    // Clear message after 1 second
+    setTimeout(() => setCopyMessage(""), 1000);
+
+    return false;
+  };
+
+  const onToggleSecret = (evt: React.MouseEvent) => {
+    evt.preventDefault();
+
+    setShowSecret(!showSecret);
+    return false;
+  };
+
+  const onEditSecretClick = () => {
+    if (toggleSecretEditorModal && setSelectedSecret) {
+      setSelectedSecret(secret);
+      toggleSecretEditorModal();
+    }
+  };
+
+  const onDeleteSecretClick = () => {
+    if (toggleDeleteSecretModal && setSelectedSecret) {
```

Comment:
```suggestion
    evt.preventDefault();
    if (toggleDeleteSecretModal && setSelectedSecret) {
```

Reply:
I added an evt: React..... to this but it didn't fix the button switch issue